### PR TITLE
[RATE-LIMIT] Query Rate Limits by ChainID

### PR DESCRIPTION
**Pull request by sampocs**
Closes: #XXX

## Brief Changelog
* Added query to get all rate limits for a given chain
* Added channel keeper to rate limit keeper

### CODE REVIEW

**Review by antstalepresh — APPROVED** [review]

File: app/app.go [hunk 1]
```diff
@@ -421,30 +421,31 @@ func NewStrideApp(
...
+		app.IBCKeeper.ChannelKeeper,
 	)
 	ratelimitModule := ratelimitmodule.NewAppModule(appCodec, app.RatelimitKeeper)
 
 	// Create ICS4Wrapper middleware
-	rateLimitingParams := app.GetSubspace(ratelimitmoduletypes.ModuleName)
-	rateLimitingParams = rateLimitingParams.WithKeyTable(ratelimitmoduletypes.ParamKeyTable())
-	rateLimitingICS4Wrapper := ratelimitmodule.NewICS4Middleware(
+	rateLimitParams := app.GetSubspace(ratelimitmoduletypes.ModuleName)
+	rateLimitParams = rateLimitParams.WithKeyTable(ratelimitmoduletypes.ParamKeyTable())
```
Comment (antstalepresh) [inline app/app.go:438]: I think we don't need these two lines anymore as we don't use rateLimitParams anymore in app.go

File: x/ratelimit/keeper/grpc_query.go [hunk 2]
```diff
@@ -4,18 +4,24 @@ import (
...
+
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v3/modules/light-clients/07-tendermint/types"
 
 	"github.com/Stride-Labs/stride/v4/x/ratelimit/types"
 )
 
 var _ types.QueryServer = Keeper{}
 
+// Query a all rate limits
```
Comment (antstalepresh) [inline x/ratelimit/keeper/grpc_query.go:17]:
```suggestion
// Query all rate limits
```

File: x/ratelimit/keeper/msg_server_test.go [hunk 3]
```diff
@@ -73,22 +113,18 @@ func (s *KeeperTestSuite) TestMsgServer_UpdateRateLimit() {
...
-	// Mint tokens for generating channel value
-	err := s.App.BankKeeper.MintCoins(s.Ctx, minttypes.ModuleName, sdk.NewCoins(sdk.NewInt64Coin(updateRateLimitMsg.Denom, channelValue)))
-	s.Require().NoError(err)
+	// Create channel and
+	s.createChannel(channelId)
+	s.createChannelValue(denom, channelValue)
 
 	// Attempt to update a rate limit that does not exist
-	_, err = s.GetMsgServer().UpdateRateLimit(sdk.WrapSDKContext(s.Ctx), &updateRateLimitMsg)
+	_, err := s.GetMsgServer().UpdateRateLimit(sdk.WrapSDKContext(s.Ctx), &updateRateLimitMsg)
```
Comment (antstalepresh) [inline x/ratelimit/keeper/msg_server_test.go:123]: What about removing GetMsgServer function and instead, adding MsgServer field in KeeperTestSuite struct?
like similar to what you did for QueryClient

File: x/ratelimit/keeper/msg_server_test.go [hunk 4]
```diff
@@ -73,22 +113,18 @@ func (s *KeeperTestSuite) TestMsgServer_UpdateRateLimit() {
 
 	denom := updateRateLimitMsg.Denom
 	channelId := updateRateLimitMsg.ChannelId
-	channelValue := int64(100)
+	channelValue := sdk.NewInt(100)
 
-	// Mint tokens for generating channel value
-	err := s.App.BankKeeper.MintCoins(s.Ctx, minttypes.ModuleName, sdk.NewCoins(sdk.NewInt64Coin(updateRateLimitMsg.Denom, channelValue)))
-	s.Require().NoError(err)
+	// Create channel and
```
Comment (antstalepresh) [inline x/ratelimit/keeper/msg_server_test.go:118]:
```suggestion
	// Create channel and channel value
```

File: x/ratelimit/keeper/rate_limit_test.go [hunk 5]
```diff
@@ -75,15 +77,29 @@ func (s *KeeperTestSuite) TestRemoveRateLimit() {
...
 	expectedRateLimits := s.createRateLimits()
 	actualRateLimits := s.App.RatelimitKeeper.GetAllRateLimits(s.Ctx)
 	s.Require().Len(actualRateLimits, len(expectedRateLimits))
 	s.Require().ElementsMatch(expectedRateLimits, actualRateLimits, "all rate limits")
 }
 
-func (s *KeeperTestSuite) SetupCheckRateLimitTest() {
-	// Add rate limit to store
+// Adds a rate limit object to the store in preparation for the check rate limit tests
+func (s *KeeperTestSuite) SetupCheckRateLimitAndUpdatedFlowTest() {
```
Comment (antstalepresh) [inline x/ratelimit/keeper/rate_limit_test.go:102]:
```suggestion
func (s *KeeperTestSuite) SetupCheckRateLimitAndUpdateFlowTest() {
```

File: app/app.go (hunk 1 above)
Comment (sampocs) [reply app/app.go:438]: Good catch! 

Do we still need the params stuff everywhere else? (e.g. params.proto, keeper/params.go, types/params.go)?

File: x/ratelimit/keeper/msg_server_test.go (hunk 3 above)
Comment (sampocs) [reply x/ratelimit/keeper/msg_server_test.go:123]: I'm glad you mentioned this! I thought the same thing when I added the query client but forgot to go back and try it

I had assumed we set it up with the Getter intentionally for stakeibc, but I can't seem to recall why we needed to do that.

File: x/ratelimit/keeper/msg_server_test.go (hunk 4 above)
Comment (sampocs) [reply x/ratelimit/keeper/msg_server_test.go:118]: fixed

File: x/ratelimit/keeper/msg_server_test.go (hunk 3 above)
Comment (antstalepresh) [reply x/ratelimit/keeper/msg_server_test.go]: Ok

**Review by antstalepresh — APPROVED** [review]

**Pull request merged**